src/app/duration-picker/duration-picker.component.ts

```diff
@@ -232,13 +311,4 @@ export class DurationPickerComponent implements OnInit, ControlValueAccessor {
     this.onTouched();
     this.onChange(this.value);
   }
-
-  // Attach all the changes received in the options object
-  attachChanges(options: any): void {
-    Object.keys(options).forEach(param => {
-      if (this.config.hasOwnProperty(param)) {
-        (this.config)[param] = options[param];
-      }
-    });
-  }
```

Comment:
???

Reply:
I chose to change this for the following reasons:

1) It could have unexpected outputs. Consider the following:

```
this.options = { showNegative: true };
setTimeout(() => this.options = { showSeconds: false }, 1000);
```

The way it was before it would actually end up the union of those two options and `showNegative` would remain true even if the options were replaced.

2) The new approach favors immutability over mutability
3) The new approach is type safe and doesn't use an any type and will work nicely with AOT